openedx/core/djangoapps/agreements/views.py

```diff
@@ -119,3 +121,43 @@ def post(self, request, course_id):
         signature = create_integrity_signature(username, course_id)
         serializer = IntegritySignatureSerializer(signature)
         return Response(serializer.data)
+
+
+class LTIPIISignatureView(AuthenticatedAPIView):
+    """
+    Endpoint for a LTI PII Signature
+    /lti_pii_signature/{course_id}
+
+    HTTP POST
+        * Do not create an LTIPIISignature record if the feature is not enabled via the CourseWaffleFlag
+        * If an LTI PII signature does not exist for the user + course, creates one and
+          returns it. If one does exist, returns the existing signature.
+    """
+
+    def post(self, request, course_id):
+        """
+        Create an LTI PII signature for the requesting user and course. If a signature
+        already exists, returns the existing signature instead of creating a new one.
+
+        /api/agreements/v1/lti_pii_signature/{course_id}
+
+        Example response:
+            {
+                username: "janedoe",
+                course_id: "org.2/course_2/Run_2",
+                created_at: "2021-04-23T18:25:43.511Z"
+            }
+        """
+        if not settings.FEATURES.get('ENABLE_LTI_PII_ACKNOWLEDGEMENT'):
+            return Response(
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        serializer = LTIPIISignatureSerializer(data=request.data)
+        print("request data", request.data)
+        if serializer.is_valid():
```

Comment:
Great job with making the test for this view! Would you also be able to:

1. Add an else statement here that makes the returned Response contain a [status of 400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) (I think), which would capture the case where the serializer is invalid.
2. Make a test for that case.

I'd recommend creating a variable called `status` before the if/else statement, and setting that variable's value within each of the if and else blocks, the adding that status to the returned Response.